delta-api/src/main/java/io/cdap/delta/api/SourceProperties.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.delta.api;
+
+import java.util.Objects;
+
+/**
+ * Represents properties of source in replicator pipeline that are exposed to the targets.
+ */
+public class SourceProperties {
+
+  /**
+   * Enum describing ordering of events.
+   */
+  public enum Ordering {
+    ORDERED,
+    UN_ORDERED
+  }
+
+  private final Ordering ordering;
+  private final RowIdKey rowIdKey;
+
+  private SourceProperties(Ordering ordering, RowIdKey rowIdKey) {
+    this.ordering = ordering;
+    this.rowIdKey = rowIdKey;
+  }
+
+  public Ordering getOrdering() {
+    return ordering;
+  }
+
+  public RowIdKey getRowIdKey() {
+    return rowIdKey;
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+    SourceProperties that = (SourceProperties) o;
+    return ordering == that.ordering &&
+      rowIdKey.equals(that.rowIdKey);
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(ordering, rowIdKey);
+  }
+
+  /**
+   * Builder for a source properties.
+   */
+  public static class Builder {
+    private Ordering ordering;
+    private RowIdKey rowIdKey;
+
+    public Builder() {
+      ordering = Ordering.ORDERED;
+    }
+
+    /**
+     * Specify the ordering of the events generated by source.
```

Comment:
nit: Sets (to keep it consistent with comments of other set method)

Reply:
fixed.